### ISSUE TITLE
Sesssionid is zero . 

### ISSUE THREAD

**Issue by WaqasYousafy786** [google/ExoPlayer#7194]
### [REQUIRED] Issue description
I am using exoplayer for my mediaplayer app . i want the sessionid for equalizer to pass . If i click on next button of player controls for playlist next song everytime sessionid  is zero and equalizer not working . 
If song changed by itself then equlizer works.

### [REQUIRED] Version of ExoPlayer being used
I am using exoplayer2.11.x

### [REQUIRED] Device(s) and version(s) of Android being used
I am using lollipop

**Comment by AquilesCanta** [comment]
@kim-vde Can you look into this? Also, could you update the issue title to describe what the issue is if you have more context?

**Comment by waqasyousafy** [comment]
@kim-vide are you here

**Comment by ojw28** [comment]
@waqasyousafy - The way you keep pinging these bugs is not appropriate. You filed it 2 hours ago. Please stop.

**Comment by waqasyousafy** [comment]
ok

**Comment by kim-vde** [comment]
I tried to reproduce the problem but I couldn't. I have an equalizer and it is still working if I go to the next item in a playlist.

Could you provide reproduction steps (for example, sample code and media)? Could you also, if possible, debug and check the value of `audioSessionId` in `DefaultAudioSink`?

**Comment by waqasyousafy** [comment]
i use this code to get the sessionid 
sessionId = mExoPlayerView.getPlayer().getAudioComponent().getAudioSessionId();
i have attached it to this code 

     if (mExoPlayerView.getPlayer().getAudioComponent() != null) {
                    sessionId = mExoPlayerView.getPlayer().getAudioComponent().getAudioSessionId();
                equalizerFragment = EqualizerFragment.newBuilder()
                        .setAccentColor(Color.parseColor("#4caf50"))
                        .setAudioSessionId(sessionId)
                        .build();
                getSupportFragmentManager().beginTransaction()
                        .replace(R.id.eqFrame, equalizerFragment)
                        .addToBackStack("equalizer")
                        .commit();
            }
I got zero every time when i press next button of player

**Comment by waqasyousafy** [comment]
I am trying my best to resolve this but i got zero at everytime as i mentioned bove . please help me . @kim-vde

**Comment by waqasyousafy** [comment]
I have solved my problem guys . Very happy on it . Thank you not to you all .

**Comment by kim-vde** [comment]
I will close this bug. If you want to receive help from our team in the future, please change your tone. We do not have the capacity to always answer issues immediately.

**Comment by RowlandOti** [comment]
Hi, @waqasyousafy  how did you resolve the issue. I am facing similar issues with a visualiser where the audioSessioId sometimes returns 0. Is it that it is set a little bit later sometimes when clicking next song?

**Comment by RowlandOti** [comment]
Had to use:

```java
player?.addAnalyticsListener(object: AnalyticsListener {
                override fun onAudioSessionId(eventTime: EventTime, audioSessionId: Int) {
                    super.onAudioSessionId(eventTime, audioSessionId)
                }
                
            })
```

as it actually gets called when the session id gets set.  Just calling `player?.audioSessionId` would leave you with `C.AUDIO_SESSION_ID_UNSET` which is 0.

**Comment by ojw28** [comment]
We made some fixes to `audioSessionId` handling in `2.13.0`, which largely resolved this issue. There is one remaining issue, which will be fixed in `2.13.2`, which is scheduled for release over the next week or two.
